전민주/Week 4/HW/app/src/main/java/com/mingmingmon/week4/LoginActivity.kt

```diff
@@ -0,0 +1,89 @@
+package com.mingmingmon.week4
+
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import com.mingmingmon.week4.databinding.ActivityLoginBinding
+
+class LoginActivity : AppCompatActivity(){
+    private lateinit var binding: ActivityLoginBinding
+    private lateinit var activityResultLancher: ActivityResultLauncher<Intent>
+    var user: User? = null
+
+    override fun onCreate(savedInstanceState : Bundle?){
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate((layoutInflater))
+        setContentView(binding.root) // 알맞은 xml 파일과 바인딩
+
+        activityResultLancher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
+            result ->
+            if(result.resultCode == Activity.RESULT_OK){
+                val data: Intent? = result.data
+                if (data != null) {
+                    user = data?.getParcelableExtra("USER", User::class.java)
+                    binding.activityLoginEtID.setText(user!!.id)
+                    binding.activityLoginEtPassword.setText(user!!.password)
```

Comment:
상위 data를 검증하며 null이 아님을 확인했더라도, !!는 가능한 쓰지 않는 것이 좋습니다!
```suggestion
                if (data != null) {
                    user = data?.getParcelableExtra("USER", User::class.java)?: return
                    binding.activityLoginEtID.setText(user.id)
                    binding.activityLoginEtPassword.setText(user.password)
           
```